### PR TITLE
node e2e: kubelet should use kubenet

### LINKED ISSUE #27498: unregister_netdevice kernel race on coreos image
Doesn't seem to haven been caused by my change: https://storage.cloud.google.com/kubernetes-jenkins/pr-logs/pull/27083/node-pull-build-e2e-test/10720/build-log.txt

```
Summarizing 3 Failures:

[Fail] [k8s.io] Downward API [It] should provide pod name and namespace as env vars [Conformance] 
/var/lib/jenkins/workspace/node-pull-build-e2e-test/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2022

[Fail] [k8s.io] Downward API [It] should provide pod IP as an env var 
/var/lib/jenkins/workspace/node-pull-build-e2e-test/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2022

[Fail] [k8s.io] Downward API [It] should provide container's limits.cpu/memory and requests.cpu/memory as env vars 
/var/lib/jenkins/workspace/node-pull-build-e2e-test/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2022

Ran 19 of 22 Specs in 1130.000 seconds
FAIL! -- 16 Passed | 3 Failed | 0 Pending | 3 Skipped --- FAIL: TestE2eNode (1130.00s)
FAIL

Ginkgo ran 1 suite in 18m50.080565864s
Test Suite Failed
```

```
[k8s.io] Downward API
/var/lib/jenkins/workspace/node-pull-build-e2e-test/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:660
  should provide container's limits.cpu/memory and requests.cpu/memory as env vars [It]
  /var/lib/jenkins/workspace/node-pull-build-e2e-test/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/downward_api_test.go:130

  Expected error:
      <*errors.errorString | 0xc8207f9770>: {
          s: "gave up waiting for pod 'downward-api-63e732e1-334a-11e6-b754-42010af00048' to be 'success or failure' after 5m0s",
      }
      gave up waiting for pod 'downward-api-63e732e1-334a-11e6-b754-42010af00048' to be 'success or failure' after 5m0s
  not to have occurred

  /var/lib/jenkins/workspace/node-pull-build-e2e-test/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2022
```

### PR DESCRIPTION
**Pull request by yujuhong**
This fixes #27498

### CODE REVIEW

**Comment by yujuhong** [comment]
/cc @pwittrock @freehan @kubernetes/sig-node 

Will need to update the trusty images

**Comment by freehan** [comment]
I think you accidentally include someone else's commit.

**Comment by ixdy** [comment]
@k8s-bot test this please, issue #IGNORE

(Sorry, temporarily broke Jenkins.)

File: test/e2e_node/environment/setup_host.sh [hunk 1]
```diff
@@ -62,6 +62,25 @@ if [ $? -ne 0 ]; then
...
 fi
 
+# Install nsenter for ubuntu images
+cat /etc/*-release | grep "ID=ubuntu"
+if [ $? -eq 0 ]; then
+  if ! which nsenter > /dev/null; then
+     echo "Do not find nsenter. Install it."
+     mkdir -p /tmp/nsenter-install
+     cd /tmp/nsenter-install
+     curl https://www.kernel.org/pub/linux/utils/util-linux/v2.24/util-linux-2.24.tar.gz | tar -zxf-
```
Comment (freehan) [inline test/e2e_node/environment/setup_host.sh]: I wonder if there is easier way to install nsenter. But it seems like this is the way to go.

**Comment by freehan** [comment]
LGTM. pending on tests.

File: test/e2e_node/environment/setup_host.sh (hunk 1 above)
Comment (yujuhong) [inline test/e2e_node/environment/setup_host.sh]: I mostly just copied this from `gce/trusty/configure-helper.sh` and verified that on a trusty instance.

Comment (euank) [inline test/e2e_node/environment/setup_host.sh]: If we want to trust jpetazzo / Dockerhub (arbitrary code execution as a service :) there's https://github.com/jpetazzo/nsenter#how-do-i-install-nsenter-with-this

**Comment by yujuhong** [comment]
Shortened the image names to stop gcloud from complaining.

**Comment by euank** [comment]
Do we need to also set `--hairpin-mode=hairpin-veth`?

**Comment by yujuhong** [comment]
> Do we need to also set --hairpin-mode=hairpin-veth?

@freehan should we do that or leave it to `promiscuous-bridge`?

**Comment by yujuhong** [comment]
BTW, I created new images and all tests passed. PTAL!

**Comment by yujuhong** [comment]
Marking p1 to deflake the node e2e suite.

**Comment by freehan** [comment]
> Do we need to also set --hairpin-mode=hairpin-veth?

Isn't this the way it was? If not, let us do it.

File: test/e2e_node/environment/setup_host.sh (hunk 1 above)
Comment (yujuhong) [inline test/e2e_node/environment/setup_host.sh]: Not that I don't trust jeptazzo, but I've already run the script and created the new images. Let's try to stop the bleeding first :-)

**Comment by yujuhong** [comment]
> Isn't this the way it was? If not, let us do it

What's the reason for that? I'm asking because the we set `--hairpin-mode=promiscuous-bridge` in a regular kubernetes cluster

**Comment by yujuhong** [comment]
Discussed with @freehan offline. He's quite confident that "hairpin-veth" + "kubenet" should solve the problem even though this combination is still in the testing stage (hence not the default). I pushed the commits to switch to using "hairpin-veth" to help test his fix. If we are still seeing the kernel bug, we'll switch back to "promiscuous-bridge"

**Comment by freehan** [comment]
Thanks!

**Comment by euank** [comment]
LGTM on green

**Comment by yujuhong** [comment]
Node e2e suite passed. Applied lgtm based on previous comments.

**Pull request merged**